### PR TITLE
Add support for external links in nav files

### PR DESCRIPTION
**Pull request by mheap**
The `absolute_url` parameter in nav files was not being respected. This PR adds support

(Broken links will fail for this PR. It's an issue with the base branch that is being worked on)

Did you sign your commit? Yes

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)? Yes

### CODE REVIEW

**Comment by lahabana** [comment]
@fabianrbz this has been done right? I think we can close this

**Comment by fabianrbz** [comment]
@lahabana I haven't done anything related to this. If we need this, we should review, test and merge it

**Comment by lahabana** [comment]
Sorry read too quickly and thought it was your external link rendering and target=blank work

**Comment by mheap** [comment]
Switched to `master` and rebased

**Comment by lahabana** [comment]
@mheap I've just pushed a link to the changelog as an example and it doesn't look like it worked out.

**Comment by mheap** [comment]
@lahabana Seems to be working for me on https://deploy-preview-1091--kuma.netlify.app/docs/dev/

**Comment by lahabana** [comment]
Yes weird! I tried locally but it wasn't working I guess I had some stale state :) 
Anyway to add a `target="_blank"` on these?

File: app/_includes/sidebar_link.html [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+{% else %}{% assign url = '/docs/' | append: include.version | append: item.url %}
+{% endif %}
 <a href="{{ url }}" class="sidebar-link {% if current_page == url %} active {% endif %}">{{ item.text or item.title }}</a>
 
 {% if item.items and item.items.size > 0 %}
   <ul class="sidebar-sub-headers {% if current_page != url %} hidden {% endif %}">
     {% for sub_item in item.items %}
       <li class="sidebar-sub-header">
-        {% assign url = '/docs/' | append: include.version | append: sub_item.url %}
+        {% if sub_item.absolute_url %}{% assign url = sub_item.url %}
```
Comment (lahabana) [inline app/_includes/sidebar_link.html:13]: Would it make sense to just have `sub_item.absolute_url` just be the value or use the fact that url starts with a scheme?

Comment (mheap) [reply app/_includes/sidebar_link.html:13]: This implementation matches how docs.konghq.com does it

File: app/_includes/sidebar_link.html [hunk 2]
```diff
@@ -1,17 +1,20 @@
 {% assign item = include.item %}
 {% assign current_page =  page.url | remove: '.html' %}
 
-<!-- "#TODO: handle external links" -->
-{% assign url = '/docs/' | append: include.version | append: item.url %}
-<a href="{{ url }}" class="sidebar-link {% if current_page == url %} active {% endif %}">{{ item.text or item.title }}</a>
+{% if item.absolute_url %}{% assign url = item.url %}
+{% else %}{% assign url = '/docs/' | append: include.version | append: item.url %}
+{% endif %}
+<a href="{{ url }}" class="sidebar-link {% if current_page == url %} active {% endif %}{% if item.absolute_url %} target="_blank"{% endif %}">{{ item.text or item.title }}</a>
```
Comment (lahabana) [inline app/_includes/sidebar_link.html:7]:
```suggestion
<a href="{{ url }}" class="sidebar-link {% if current_page == url %} active {% endif %}"{% if item.absolute_url %} target="_blank"{% endif %}">{{ item.text or item.title }}</a>
```

File: app/_includes/sidebar_link.html [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
   <ul class="sidebar-sub-headers {% if current_page != url %} hidden {% endif %}">
     {% for sub_item in item.items %}
       <li class="sidebar-sub-header">
-        {% assign url = '/docs/' | append: include.version | append: sub_item.url %}
+        {% if sub_item.absolute_url %}{% assign url = sub_item.url %}
+        {% else %}{% assign url = '/docs/' | append: include.version | append: sub_item.url %}
+        {% endif %}
 
-        <a class="sidebar-link" href="{{ url }}">
+        <a class="sidebar-link" href="{{ url }}"{% if sub_item.absolute_url %} target="_blank"{% endif %}>
```
Comment (lahabana) [inline app/_includes/sidebar_link.html:17]:
```suggestion
        <a class="sidebar-link" href="{{ url }}"{% if sub_item.absolute_url %} target="_blank"{% endif %}>
```

**Review by lahabana — APPROVED** [review]
Once the quote issue is fixed LGTM!

**Comment by slonka** [comment]
Something is not right, preview link is broken - 
image

:/

**Comment by slonka** [comment]
thanks @mheap 🎉

**Pull request merged**